deepchecks/vision/vision_data/format_validators.py

```diff
@@ -0,0 +1,208 @@
+        if np.max(single_image_pred[:, 5] % 1) > 0:
+            raise ValidationError(f'Class_id must be a positive integer. {error_msg}')
+    elif task_type == TaskType.SEMANTIC_SEGMENTATION and single_image_pred is not None:
+        if len(single_image_pred.shape) != 3:
+            raise ValidationError('Semantic segmentation prediction per image must be a 3D array of shape (C, H, W),'
+                                  'where H and W are the height and width of the corresponding image, and C is the '
+                                  'number of classes that can be detected.')
+        if not 0.99 < np.sum(single_image_pred[:, 0][:, 0]) < 1.01:
+            raise ValidationError('Semantic segmentation prediction per pixel per image should be probabilities per '
+                                  'each possible class')
+
+
+def _validate_predictions_label_common_format(name, data, task_type: TaskType):
+    """Validate that the data is in the required format and returns a non-empty sample in numpy format."""
+    name_plural = name + 's'
+    if task_type == TaskType.OTHER and data is not None:
+        raise ValidationError(f'The task type is OTHER, but {name_plural} were provided.')
+    try:
+        _ = data[0]
+        data = sequence_to_numpy(data)
+    except IndexError as err:
+        raise ValidationError(f'The batch {name_plural} must be a non empty iterable.') from err
+    except TypeError as err:
+        raise ValidationError(f'The batch {name_plural} must be an iterable, received {type(data)}') from err
+
+    sample_idx = 0
+    while data[sample_idx] is None or (isinstance(data[sample_idx], np.ndarray) and data[sample_idx].shape[0] == 0):
+        sample_idx += 1
+        if sample_idx == len(data):
+            return None  # No data to validate
+
+    if task_type == TaskType.CLASSIFICATION:
+        return data[sample_idx]  # for classification, the data is a single number no need to validate shape
+    try:
+        sample_shape = data[sample_idx].shape
+    except AttributeError as err:
+        raise ValidationError(f'{name} for {task_type.value} per image must be a multi dimensional array.') from err
+
+    if task_type == TaskType.OBJECT_DETECTION and len(sample_shape) not in (1, 2):
+        raise ValidationError(f'{name} for object detection per image must be a 2D array. Found shape {sample_shape}')
+
+    return data[sample_idx]
+
+
+def validate_additional_data_format(additional_data_batch):
+    """Validate that the data is in the required format.
+
+    Parameters
+    ----------
+    additional_data_batch
+        The additional data of the batch. Result of VisionData's batch_to_additional_data
+    Raises
+    ------
+    DeepchecksValueError
+        If the images doesn't fit the required deepchecks format.
+    """
+    if not hasattr(additional_data_batch, '__getitem__'):
+        raise ValidationError('The batch additional_data must be an iterable.')
```

Comment:
__getitem__ doesn't mean iterable
if you want iterable do isinstance(stuff, t.Iterable)

Reply:
iterable != sequence

Reply:
ok, changes